File: packages/core/tests/fixtures/fixture.pas
{ Captures 'unit' (Package). }
unit MyPascalUnit;

interface

type
  TMyEnum = (
    { Captures 'declEnumValue' (EnumMember). }
    EnumAlpha,
    { Captures 'declEnumValue' (EnumMember). }
    EnumBeta
  );

  { Captures 'declType' (Type). }
  TMyPascalClass = class
  private
    { Captures 'declField' (Field). }
    FMyField: Integer;
  public
    { Captures 'defProc' (Method). }
    procedure DoSomething;
    { Captures 'defProc' (Method) with massive string. }
    procedure MassiveStringProcedure;
  end;

const
  { Captures 'declConst' (Field). }
  MY_CONST = 42;

var
  { Captures 'declVar' (Field). }
  GlobalVar: Integer;

implementation

procedure TMyPascalClass.DoSomething;
begin
  FMyField := 1;
end;

procedure TMyPascalClass.MassiveStringProcedure;
var
  MassiveString: string;
begin
  MassiveString := 'This string is designed to be exceedingly long to force the chunker to split by size. ' +
  'Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] ' +
  'Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] ' +
  'Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] ' +
  'Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] ' +
  'Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] ' +
  'Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] ' +
  'Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] ' +
  'Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX]';
end;

{ Captures 'initialization' (Initializer). }
initialization
  GlobalVar := 0;

{ Captures 'finalization' (Initializer). }
finalization
  GlobalVar := -1;

end.

File: packages/core/tests/fixtures/fixture.py
"""Module-level docstring."""

def my_decorator(func):
    """Captures 'function_definition' (Method)."""
    return func

# Captures 'decorator' (AnnotationElement)
@my_decorator
class MyPythonClass:
    """Captures 'class_definition' (Type)."""
    
    # Captures 'expression_statement (assignment)' (Field)
    my_field = 42
    
    def __init__(self):
        """Captures 'function_definition' named '__init__' (Initializer)."""
        self.value = 1

    # Captures 'decorator' with name 'property' to trigger (Accessor)
    @property
    def my_property(self):
        """Captures 'decorated_definition' (Accessor)."""
        return self.value

    def massive_string_function(self):
        """Captures 'function_definition' (Method) testing HARD_CAP_BYTES fallback."""
        massive_string = (
            "This string is designed to be exceedingly long to force the chunker to split by size. "
            "Because this string literal has no internal statement nodes, Tree-sitter's fallback kicks in. "
            "Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] "
            "Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] "
            "Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] "
            "Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] "
            "Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] "
            "Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] "
            "Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX] "
            "Padding: [XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX]"
        )
